packages/manager/modules/pci/src/projects/project/instances/instances.service.js

```diff
@@ -485,9 +496,19 @@ export default class PciProjectInstanceService {
   }
 
   getInstancePrice(projectId, instance, catalogEndpoint) {
-    return this.CucPriceHelper.getPrices(projectId, catalogEndpoint)
-      .then((prices) => {
-        const price = prices[instance.planCode];
+    return this.getCatalog(catalogEndpoint, this.coreConfig.getUser())
+      .then((catalog) => {
+        const instanceAddon = catalog.addons.find(
+          (a) => a.planCode === instance.planCode,
+        );
+        const price = this.CucPriceHelper.transformPrice(
+          instanceAddon.pricings.find(
```

Comment:
`instanceAddon` is possibly `undefined`

Reply:
It isn't possible, if instance has a plancode which isn't in the catalog that means there is a free instance